Automattic/woocommerce-payments
# Show empty state treatment only when WCPay not connected

Fixes #2511

Huge kudos to @htdat for discovering (https://github.com/Automattic/woocommerce-payments/issues/2511#issuecomment-886051004) the likely cause of the issue.

#### Changes proposed in this Pull Request

If empty state experimental treatment (added in https://github.com/Automattic/woocommerce-payments/pull/1963) is active, it appears that Transactions and Deposits are currently not being shown. The change in https://github.com/Automattic/woocommerce-payments/commit/3d1bc5a02676c8d21bbb3bc3d74eebc0b29735e3 is to always show them when WCPay is connected.

Also, https://github.com/Automattic/woocommerce-payments/commit/2f24127eb990fd79ed2bcf6d56ba3d534f1e7774 ensures that the redirect to the onboarding screen (added in https://github.com/Automattic/woocommerce-payments/pull/2275) occurs only when not in treatment mode.

#### Testing instructions

First, go into treatment mode:

I tested this only in a local way by returning true from `is_in_treatment_mode`, and verifying the proper view with console statements in the `TransactionsPage` and `DepositsPage` components: `console.log( treatmentExperience === defaultExperience ? 'default table' : 'new empty state' );`

I would encourage testing by assigning the `treatment` variations as described in https://github.com/Automattic/woocommerce-payments/pull/1963.

Next, navigate to Deposits and Transactions screens on a connected site, and verify that the **default table view appears** (with data, or "No data to display" message) and the **"Finish setup" button does not**.

Also, test with a disconnected site (or set `wcpaydev_force_disconnected` option to true e.g. via WCPay dev tools) and verify that the "Finish setup" button appears on Deposits and Transactions screens, even on new page load.

-------------------


**Not adding a changelog entry** mainly because I'm not sure whether this change will be rebased or cherry-picked into release branch, but one should be added prior to merge or release.
